lib/Cake/Model/Datasource/DboSource.php

```diff
@@ -1162,7 +1160,20 @@ public function read(Model $Model, $queryData = array(), $recursive = null) {
 					continue;
 				}
 
-				if ($noAssocFields) {
+				if (!empty($queryData['fields'])) {
+					foreach ($queryData['fields'] as $key => $field) {
+						$fieldExploded = explode('.', $field);
+						if (count($fieldExploded) === 2 && str_replace('`', '', $fieldExploded[0]) === $LinkModel->alias) {
```

Comment:
You might want to consider using a regexp, as it might be a simpler way to extract the alias/field out of the column name.